{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get Code References from papers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook:\n",
    "\n",
    " * Uses getpapers externally to download fulltext of all papers in EuPMC which contain github URLs\n",
    " * Textmines each paper fulltext and extract occurences of GitHub URLs\n",
    " * Outputs a data structure of the form: paper_DOI {{github_url: \"http://github.com/blah/blah\"}...}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from lxml import etree\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use getpapers to download fulltext of papers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We currently do this outside of the notebook, and assume that the files are available locally.\n",
    "\n",
    "The command we are using is:\n",
    "\n",
    ">getpapers --query 'github' -x --limit 100 -o data\n",
    "\n",
    "which queries EuPMC for all papers containing the term 'github' and returns the full text of the first 100 papers matching this into the directory 'data'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Textmine each paper"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### File locations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Directory containing the data\n",
    "data_dir = '../data'\n",
    "\n",
    "# File containing the list of matching papers\n",
    "matching_papers = data_dir + '/' + 'eupmc_fulltext_html_urls.txt'\n",
    "\n",
    "# Name of the Content Mine results file in each paper subdirectory\n",
    "contentmine_results = 'eupmc_result.json'\n",
    "\n",
    "# Name of the Content Mine full text xml paper dump in each paper subdirectory\n",
    "fulltext_xml = 'fulltext.xml'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Object for building the JSON output file containing the dictionary of papers and URLs to repositories\n",
    "\n",
    "dict_of_papers = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the list of subdirectories dumped by ContentMine\n",
    "\n",
    "papers = []\n",
    "\n",
    "# For each line in matching_papers, strip the start of the line (http://europepmc.org/articles/)\n",
    "\n",
    "with open(matching_papers, 'r') as f:\n",
    "  for line in f:\n",
    "     terms = line.split(\"/\")\n",
    "     papers.append(terms[-1].rstrip())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: File does not appear to exist.\n",
      "Error: File does not appear to exist.\n"
     ]
    }
   ],
   "source": [
    "# For each paper\n",
    "\n",
    "for paper_dir in papers:\n",
    "\n",
    "    # Read in the JSON file and get the DOI\n",
    "    \n",
    "    filename = data_dir + '/' + paper_dir + '/' + contentmine_results\n",
    "    \n",
    "    try:\n",
    "        with open(filename, 'r') as f:\n",
    "            paper_json = json.load(f)\n",
    "            # Get the DOI\n",
    "            paper_doi = paper_json['doi'][0]\n",
    "    except IOError:\n",
    "        print(\"Error: File does not appear to exist.\")\n",
    "    \n",
    "    # Read in the XML full text and mine for the github URLs\n",
    "\n",
    "    fulltext_file = data_dir + '/' + paper_dir + '/' + fulltext_xml\n",
    " \n",
    "    gh_dict = {}\n",
    "    gh_urls = []\n",
    "\n",
    "    try:\n",
    "        with open(fulltext_file, 'r') as f:\n",
    "            data = f.read()\n",
    "            urls = re.findall(r'(https?://\\S+)(?=\\\")', data)\n",
    "            for url in urls:\n",
    "                if re.match(r'https?://github.com', url):\n",
    " #                   print(url)\n",
    "                    gh_urls.append(url) \n",
    "    except IOError:\n",
    "        print(\"Error: File does not appear to exist.\")\n",
    "\n",
    "    gh_dict['github'] = gh_urls\n",
    "\n",
    "    dict_of_papers[str(paper_doi)] = gh_dict    \n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Output data structure"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"10.1002/1873-3468.12684\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/epierson9/ZIFA\",\n",
      "            \"https://github.com/dgrun/RaceID\",\n",
      "            \"https://github.com/JohnReid/DeLorean\",\n",
      "            \"https://github.com/kieranrcampbell/pseudogp\",\n",
      "            \"https://github.com/ManuSetty/wishbone\",\n",
      "            \"https://github.com/Teichlab/scrnatb\",\n",
      "            \"https://github.com/ManuSetty/wishbone\",\n",
      "            \"https://github.com/SheffieldML/GPclust\",\n",
      "            \"https://github.com/RGLab/MAST\",\n",
      "            \"https://github.com/tallulandrews/M3Drop\",\n",
      "            \"https://github.com/kdkorthauer/scDD\",\n",
      "            \"https://github.com/catavallejos/BASiCS\",\n",
      "            \"https://github.com/PMBio/scLVM\",\n",
      "            \"https://github.com/brentp/combat.py/blob/master/R-combat.R\",\n",
      "            \"https://github.com/lengning/OEFinder\",\n",
      "            \"https://github.com/drisso/RUVSeq\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1002/pmic.201700244\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/PRIDE-Utilities/ols-client\",\n",
      "            \"https://github.com/PRIDE-Toolsuite/ols-dialog\",\n",
      "            \"https://github.com/PRIDE-Utilities/ols-client\",\n",
      "            \"https://github.com/PRIDE-Toolsuite/ols-dialog\",\n",
      "            \"https://github.com/compomics/colims\",\n",
      "            \"https://github.com/PRIDE-Utilities/ols-client\",\n",
      "            \"https://github.com/PRIDE-Toolsuite/ols-dialog\",\n",
      "            \"https://github.com/proteomexchange/px-submission-tool\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1016/j.cels.2017.07.007\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/cschlaffner/PoGo\",\n",
      "            \"https://github.com/cschlaffner/PoGoGUI\",\n",
      "            \"https://github.com/cschlaffner/FileConverter\",\n",
      "            \"https://github.com/cschlaffner/TrackHubGenerator\",\n",
      "            \"https://github.com/FenyoLab/PGx\",\n",
      "            \"https://github.com/FenyoLab/PGx\",\n",
      "            \"https://github.com/cschlaffner/PoGo\",\n",
      "            \"https://github.com/cschlaffner/PoGoGUI\",\n",
      "            \"https://github.com/cschlaffner/FileConverter\",\n",
      "            \"https://github.com/cschlaffner/TrackHubGenerator\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1016/j.csbj.2017.10.002\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/aquaskyline/LRSIM\",\n",
      "            \"https://github.com/aquaskyline/LRSIM\",\n",
      "            \"https://github.com/aquaskyline/LRSIM/releases/tag/1.0\",\n",
      "            \"https://github.com/aquaskyline/LRSIM/tree/master/test\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1016/j.dib.2017.07.049\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/OpenNFT/OpenNFT_Demo/releases\",\n",
      "            \"https://github.com/OpenNFT/OpenNFT_Demo/releases\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1021/acs.analchem.7b03544\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/compomics/meta-proteome-analyzer\",\n",
      "            \"https://github.com/compomics/meta-proteome-analyzer\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/s41467-018-03297-7\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/YeatmanLab/AFQ-Browser_data\",\n",
      "            \"https://github.com/YeatmanLab/AFQ-Browser_data/blob/master/AFQ-Browser_ALSexample/Reproducing-Sarica2017-Figure3.ipynb\",\n",
      "            \"https://github.com/YeatmanLab/AFQ-Browser_data/blob/master/AFQ-Browser_ALSexample/Figure6.ipynb\",\n",
      "            \"https://github.com\",\n",
      "            \"https://github.com/YeatmanLab/AFQ-Browser\",\n",
      "            \"https://github.com/YeatmanLab/AFQ-Browser/\",\n",
      "            \"https://github.com/yeatmanlab/AFQ-Browser_data\",\n",
      "            \"https://github.com/yeatmanlab/AFQBrowser-demo/\",\n",
      "            \"https://github.com/yeatmanlab/AFQ-Browser-MSexample/\",\n",
      "            \"https://github.com/yeatmanlab/Sarica_2017\",\n",
      "            \"https://github.com/mrdoob/three.js/\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/s41467-018-03311-y\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/galaxyproteomics/tools-galaxyp\",\n",
      "            \"https://github.com/yafeng/proteogenomics_python\",\n",
      "            \"https://github.com/yafeng/proteogenomics_python\",\n",
      "            \"https://github.com/yafeng/SpectrumAI\",\n",
      "            \"https://github.com/yafeng/proteogenomics_python/scam_bams.py\",\n",
      "            \"https://github.com/yafeng/proteogenomics_python/calculate_phastcons.py\",\n",
      "            \"https://github.com/yafeng/proteogenomics_python/calculate_phylocsf.py\",\n",
      "            \"https://github.com/lehtiolab/proteogenomics-analysis-workflow\",\n",
      "            \"https://github.com/yafeng/proteogenomics_python\",\n",
      "            \"https://github.com/yafeng/SpectrumAI\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/s41598-017-06174-3\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/Gregor-Mendel-Institute/small-RNA-spike-ins/methods.sRNA.spike.in.design/methods.sRNA.spike.in.design.step1.py\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/small-RNA-spike-ins/methods.sRNA.spike.in.design/methods.sRNA.spike.in.design.step2.endogenous.miRNAs.py\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/small-RNA-spike-ins/methods.sRNA.spike.in.design/sRNA.spike.in.design.step2.py\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/small-RNA-spike-ins/methods.sRNA.spike.in.design/examine.MFE.distributions.R\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/small-RNA-spike-ins/methods.data.analysis/small.RNA.Seq/getMirsSirsAndTas_v03.py\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/small-RNA-spike-ins/methods.data.analysis/master_genes_Araport.txt\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/small-RNA-spike-ins/\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/s41598-017-15885-6\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/andlaw1841/Andy-s-Algorithm\",\n",
      "            \"https://github.com/andlaw1841/Andy-s-Algorithm\",\n",
      "            \"https://github.com/andlaw1841/Andy-s-Algorithm\",\n",
      "            \"https://github.com/andlaw1841/Andy-s-Algorithm\",\n",
      "            \"https://github.com/andlaw1841/Andy-s-Algorithm\",\n",
      "            \"https://github.com/andlaw1841/Andy-s-Algorithm\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/s41598-017-18257-2\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/ChrisMaherLab/INTEGRATE-Vis\",\n",
      "            \"https://github.com/ChrisMaherLab/INTEGRATE-Vis\",\n",
      "            \"https://github.com/ChrisMaherLab/INTEGRATE-Vis\",\n",
      "            \"https://github.com/ChrisMaherLab/INTEGRATE-Vis\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/s41598-018-19635-0\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/kouroshz\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/s41598-018-22181-4\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/neuropoly/axondeepseg\",\n",
      "            \"https://github.com/neuropoly/axondeepseg\",\n",
      "            \"https://github.com/neuropoly/axondeepseg\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/sdata.2017.149\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/lucventurini/mikado\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/sdata.2017.151\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/MaayanLab/clustergrammer\",\n",
      "            \"https://github.com/MaayanLab/clustergrammer-docs\",\n",
      "            \"https://github.com/MaayanLab/clustergrammer\",\n",
      "            \"https://github.com/MaayanLab/clustergrammer-py\",\n",
      "            \"https://github.com/jupyter-widgets/widget-cookiecutter\",\n",
      "            \"https://github.com/MaayanLab/clustergrammer-widget\",\n",
      "            \"https://github.com/jupyter-widgets/ipywidgets\",\n",
      "            \"https://github.com/MaayanLab/clustergrammer\",\n",
      "            \"https://github.com/MaayanLab/clustergrammer-py\",\n",
      "            \"https://github.com/MaayanLab/clustergrammer-widget\",\n",
      "            \"https://github.com/MaayanLab/clustergrammer-web\",\n",
      "            \"https://github.com/MaayanLab/CST_Lung_Cancer_Viz\",\n",
      "            \"https://github.com/MaayanLab/Cytof_Plasma_PMA\",\n",
      "            \"https://github.com/MaayanLab/CCLE_Clustergrammer\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/sdata.2017.184\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/Gregor-Mendel-Institute/SNPmatch\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/AraGeno\",\n",
      "            \"https://github.com/pypa/pip\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/SNPmatch#installation-using-pip\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/SNPmatch/blob/master/README.md\",\n",
      "            \"https://github.com/Gregor-Mendel-Institute/SNPmatch/blob/master/snpmatch/core/csmatch.py#L19\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1038/sdata.2018.11\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/npnl/ATLAS/\",\n",
      "            \"https://github.com/npnl/ATLAS/issues\",\n",
      "            \"https://github.com/BIC-MNI/minc-toolkit\",\n",
      "            \"https://github.com/npnl/ATLAS/issues\",\n",
      "            \"https://github.com/npnl/SRQL\",\n",
      "            \"https://github.com/npnl/ATLAS/\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1080/14686996.2017.1401424\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/tsudalab/ChemTS\",\n",
      "            \"http://github.com/fchollet/keras\",\n",
      "            \"https://github.com/mkusner/grammarVAE\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1093/bioinformatics/btw838\": {\n",
      "        \"github\": []\n",
      "    },\n",
      "    \"10.1093/gigascience/gix022\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/C3BI-pasteur-fr/ReGaTE\",\n",
      "            \"https://github.com/bio-tools/biotoolsSchema/\",\n",
      "            \"https://github.com/edamontology/edamontology/blob/master/HOW_TO_CONTRIBUTE.md\",\n",
      "            \"https://github.com/galaxyproject/galaxy/pull/2387\",\n",
      "            \"https://github.com/galaxyproject/galaxy/pull/2428\",\n",
      "            \"https://github.com/galaxyproject/galaxy/pull/2397\",\n",
      "            \"https://github.com/galaxyproject/galaxy/pull/2379\",\n",
      "            \"https://github.com/galaxyproject/galaxy/pull/3221\",\n",
      "            \"https://github.com/C3BI-pasteur-fr/ReGaTE\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1093/gigascience/gix025\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/cheinle/MetaLIMS\",\n",
      "            \"https://github.com/cheinle/MetaLIMS/wiki\",\n",
      "            \"https://github.com/bikalabs/bika.lims\",\n",
      "            \"https://github.com/cheinle/MetaLIMS/wiki\",\n",
      "            \"https://github.com/cheinle/MetaLIMS\",\n",
      "            \"https://github.com/cheinle/MetaLIMS\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1093/gigascience/gix100\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/\",\n",
      "            \"https://github.com/4DGenome/parallel_sequencing_lives\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1093/gigascience/gix131\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/ma-tech/eHistologyWebapp\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1093/nar/gkx1053\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/guertinlab/seqOutBias\",\n",
      "            \"https://github.com/guertinlab/kmer_mask_em\",\n",
      "            \"https://github.com/guertinlab/seqOutBias/tree/master/docs/R\",\n",
      "            \"https://github.com/guertinlab/seqOutBias\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1093/nar/gkx1132\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/reactome\",\n",
      "            \"https://github.com/reactome\",\n",
      "            \"https://github.com/reactome-pwp/diagram\",\n",
      "            \"https://github.com/reactome/\",\n",
      "            \"https://github.com/reactome-pwp/diagram\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1093/nar/gkx1152\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/Planteome\",\n",
      "            \"https://github.com/Planteome/plant-ontology\",\n",
      "            \"https://github.com/Planteome/plant-trait-ontology\",\n",
      "            \"https://github.com/Planteome/plant-experimental-conditions-ontology\",\n",
      "            \"https://github.com/pato-ontology/pato\",\n",
      "            \"https://github.com/Planteome/planteome-ncbi-taxonomy\",\n",
      "            \"https://github.com/Planteome/amigo\",\n",
      "            \"http://github.com/Planteome\",\n",
      "            \"https://github.com/biolink/biolink-api\",\n",
      "            \"https://github.com/Planteome\",\n",
      "            \"https://github.com/Planteome/plant-ontology/issues\",\n",
      "            \"https://github.com/Planteome/common-files-for-ref-ontologies/tree/master/scripts/germplasm_annotation\",\n",
      "            \"https://github.com/Planteome\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1099/mgen.0.000124\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/WGS-standards-and-analysis/datasets\",\n",
      "            \"https://github.com/andrewjpage/docker_mlst\",\n",
      "            \"https://github.com/WGS-standards-and-analysis/datasets\",\n",
      "            \"https://github.com/tseemann/mlst\",\n",
      "            \"https://github.com\",\n",
      "            \"https://github.com/sanger-pathogens/ariba\",\n",
      "            \"https://github.com/tseemann/mlst\",\n",
      "            \"https://github.com/WGS-standards-and-analysis/datasets\",\n",
      "            \"https://github.com/andrewjpage/docker_mlst\",\n",
      "            \"https://github.com/WGS-standards-and-analysis/datasets\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1126/sciadv.1700578\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/eqcorrscan\",\n",
      "            \"https://github.com/tperol/ConvNetQuake\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1128/genomea.00029-18\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/najoshi/sickle\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1128/genomea.01424-17\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/rrwick/Porechop\",\n",
      "            \"https://github.com/weizhongli/cdhit\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1128/genomea.01472-17\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/mourisl/Lighter\",\n",
      "            \"https://github.com/broadinstitute/pilon\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1128/genomea.01479-17\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/lpryszcz/pyScaf\",\n",
      "            \"https://github.com/wurmlab/flo\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1128/mbio.01650-17\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/figueroalab/Pca-genome\",\n",
      "            \"https://github.com/figueroalab/Pca-genome\",\n",
      "            \"https://github.com/figueroalab/Pca-genome\",\n",
      "            \"https://github.com/PacificBiosciences/FALCON-integrate/tree/funzip_052016\",\n",
      "            \"https://github.com/figueroalab/Pca-genome\",\n",
      "            \"https://github.com/PacificBiosciences/FALCON/tree/master/examples\",\n",
      "            \"https://github.com/nextgenusfs/funannotate\",\n",
      "            \"https://github.com/nextgenusfs/funannotate\",\n",
      "            \"https://github.com/tanghaibao/goatools\",\n",
      "            \"https://github.com/vcflib/vcflib\",\n",
      "            \"https://github.com/Adamtaranto/density-Mapr\",\n",
      "            \"https://github.com/figueroalab/Pca-genome\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1128/mbio.02275-17\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/BenjaminSchwessinger/Pst_104_E137_A-_genome\",\n",
      "            \"https://github.com/BenjaminSchwessinger/Pst_104_E137_A-_genome\",\n",
      "            \"https://github.com/BenjaminSchwessinger/Pst_104_E137_A-_genome\",\n",
      "            \"https://github.com/vcflib/vcflib\",\n",
      "            \"https://github.com/RealTimeGenomics/rtg-tools\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1128/mbio.02345-17\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_MAGs_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\",\n",
      "            \"https://github.com/MohnLab/Steroid_Degradation_Metagenomes_KRONA_charts_2017\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1177/1176934317742613\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/tomas-fer/HybPhyloMaker/\",\n",
      "            \"https://github.com/karel-brinda/ococo/\",\n",
      "            \"https://github.com/marekborowiec/AMAS/\",\n",
      "            \"https://github.com/gcollet/MstatX/\",\n",
      "            \"https://github.com/smirarab/ASTRAL/\",\n",
      "            \"https://github.com/pranjalv123/ASTRID/\",\n",
      "            \"https://github.com/smirarab/mrpmatrix/\",\n",
      "            \"https://github.com/tomas-fer/HybPhyloMaker/\",\n",
      "            \"https://github.com/bede/kindel\",\n",
      "            \"https://github.com/nylander/catfasta2phyml/\",\n",
      "            \"https://github.com/gcollet/MstatX/\",\n",
      "            \"https://github.com/marekborowiec/good_genes\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-017-1581-4\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/vuejs/vue\",\n",
      "            \"https://github.com/d3/d3\",\n",
      "            \"https://github.com/d3/d3-quadtree\",\n",
      "            \"https://github.com/DeskGen/open-guide-picker\",\n",
      "            \"https://github.com/MicrosoftResearch/Azimuth\",\n",
      "            \"https://github.com/DeskGen/open-guide-picker\",\n",
      "            \"https://github.com/MicrosoftResearch/Azimuth\",\n",
      "            \"https://github.com/vuejs/vue\",\n",
      "            \"https://github.com/d3/d3\",\n",
      "            \"https://github.com/d3/d3-quadtree\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-017-1678-9\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/uio-cels/offsetbasedgraph\",\n",
      "            \"https://github.com/uio-cels/genomicgraphcoords\",\n",
      "            \"https://github.com/uio-cels/offsetbasedgraph\",\n",
      "            \"https://github.com/uio-cels/genomicgraphcoords\",\n",
      "            \"https://github.com/uio-cels/genomicgraphcoords\",\n",
      "            \"https://github.com/uio-cels/genomicgraphcoords\",\n",
      "            \"https://github.com/uio-cels/genomicgraphcoords\",\n",
      "            \"https://github.com/vgteam/vg\",\n",
      "            \"https://github.com/GFA-spec/GFA-spec\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-017-1837-z\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/jmjuanes/vismapper/tree/master/ismapper-test\",\n",
      "            \"https://github.com/jmjuanes/vismapper\",\n",
      "            \"https://github.com/jmjuanes/vismapper/wiki\",\n",
      "            \"https://github.com/jmjuanes/vismapper/tree/master/ismapper-test\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-017-1873-8\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/shahpr/RiboViz\",\n",
      "            \"https://github.com/shahpr/RiboViz\",\n",
      "            \"https://github.com/shahpr/RiboViz\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-017-1909-0\": {\n",
      "        \"github\": [\n",
      "            \"http://github.com/jsh58/DMRfinder\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-017-1939-7\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/rhysf/Synima\",\n",
      "            \"https://github.com/rhysf/Synima\",\n",
      "            \"https://github.com/rhysf/Synima\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-017-1974-4\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/joel-lzb/CORNAS\",\n",
      "            \"https://github.com/joel-lzb/CORNAS\",\n",
      "            \"https://github.com/joel-lzb/CORNAS_Supporting_Data\",\n",
      "            \"https://github.com/joel-lzb/CORNAS\",\n",
      "            \"https://github.com/joel-lzb/CORNAS_Supporting_Data\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-018-2024-6\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/OpenGene/MutScan\",\n",
      "            \"http://github.com/sfchen/tumor-pipeline\",\n",
      "            \"https://github.com/sfchen/tumor-pipeline\",\n",
      "            \"https://github.com/OpenGene/GeneFuse\",\n",
      "            \"https://github.com/OpenGene/MutScan\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-018-2051-3\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/txje/lrc_eval\",\n",
      "            \"https://github.com/lh3/miniasm\",\n",
      "            \"https://github.com/lh3/minimap\",\n",
      "            \"https://github.com/thegenemyers/DALIGNER\",\n",
      "            \"http://github.com/holtjma/fmlrc\",\n",
      "            \"http://github.com/holtjma/msbwt\",\n",
      "            \"https://github.com/holtjma/msbwt/wiki/Converting-to-msbwt%27s-RLE-format\",\n",
      "            \"https://github.com/PacificBiosciences/DevNet/wiki/Datasets\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-018-2057-x\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/variani/lme4qtl\",\n",
      "            \"https://github.com/variani/lme4qtl\",\n",
      "            \"https://github.com/lme4/lme4\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-018-2074-9\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/bioperl/p5-bpwrapper\",\n",
      "            \"https://github.com/tlawrence3/FAST\",\n",
      "            \"https://github.com/bioperl/p5-bpwrapper\",\n",
      "            \"https://github.com/bioperl/p5-bpwrapper\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-018-2082-9\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/joheras/DecoFungi\",\n",
      "            \"https://github.com/joheras/DecolorisationImages\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12859-018-2085-6\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/mathelab/IntLIM\",\n",
      "            \"https://github.com/mathelab/IntLim/\",\n",
      "            \"https://github.com/Mathelab/NCI60_GeneMetabolite_Data\",\n",
      "            \"https://github.com/Mathelab/BreastCancerAmbs_GeneMetabolite_Data\",\n",
      "            \"https://github.com/Mathelab/IntLIM/\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12862-017-1071-3\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/josephryan/2017b_Sasson_and_Ryan\",\n",
      "            \"https://github.com/josephryan/2017b_Sasson_and_Ryan\",\n",
      "            \"https://github.com/josephryan/2017b_Sasson_and_Ryan\",\n",
      "            \"https://github.com/josephryan/2017b_Sasson_and_Ryan\",\n",
      "            \"https://github.com/josephryan/2017b_Sasson_and_Ryan\",\n",
      "            \"https://github.com/josephryan/2017b_Sasson_and_Ryan\",\n",
      "            \"https://github.com/josephryan/2017b_Sasson_and_Ryan\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12864-017-4327-9\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/yuwwu/ezTree\",\n",
      "            \"https://github.com/yuwwu/ezTree\",\n",
      "            \"https://github.com/yuwwu/ezTree\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12885-017-3689-3\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/uc-bd2k/grcalculator\",\n",
      "            \"https://github.com/uc-bd2k/grcalculator\",\n",
      "            \"https://github.com/uc-bd2k/grbrowser\",\n",
      "            \"https://github.com/uc-bd2k/grtutorial\",\n",
      "            \"https://github.com/uc-bd2k/GRmetrics\",\n",
      "            \"https://github.com/datarail/gr_metrics\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12918-017-0515-0\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/AndrasHartmann/OptPipe\",\n",
      "            \"https://github.com/AndrasHartmann/OptPipe\",\n",
      "            \"https://github.com/AndrasHartmann/OptPipe\",\n",
      "            \"https://github.com/AndrasHartmann/OptPipe\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s12977-018-0401-x\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/masato-ogishi/HANDPrediction\",\n",
      "            \"https://github.com/masato-ogishi/HANDPrediction\",\n",
      "            \"https://github.com/masato-ogishi/HANDPrediction\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13015-018-0123-6\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/yamada-kd/nepal\",\n",
      "            \"https://github.com/yamada-kd/nepal\",\n",
      "            \"https://github.com/yamada-kd/nepal\",\n",
      "            \"https://github.com/yamada-kd/nepal\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13059-017-1319-7\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/ChimeRScope/ChimeRScope/wiki\",\n",
      "            \"https://github.com/bioinfo-ut/GenomeTester4/\",\n",
      "            \"https://github.com/danielnavarrogomez/phy-mer\",\n",
      "            \"http://github.com/cbib/MICADo\",\n",
      "            \"https://github.com/lh3/minimap\",\n",
      "            \"https://github.com/marbl/MHAP\",\n",
      "            \"https://github.com/lh3/miniasm\",\n",
      "            \"https://github.com/warrenlr/LINKS/\",\n",
      "            \"https://github.com/luscinius/afcluster\",\n",
      "            \"https://github.com/mourisl/Lighter\",\n",
      "            \"https://github.com/gmarcais/Quorum\",\n",
      "            \"https://github.com/fanhuan/AAF\",\n",
      "            \"https://github.com/pdtrang/GSM\",\n",
      "            \"https://github.com/marbl/mash\",\n",
      "            \"https://github.com/jessieren/VirHostMatcher\",\n",
      "            \"https://github.com/ctlab/metafast\",\n",
      "            \"https://github.com/seqan/seqan/tree/master/apps/alf\",\n",
      "            \"https://github.com/evolbioinf/andi/\",\n",
      "            \"https://github.com/younglululu/CAFE\",\n",
      "            \"https://github.com/tderrien/FEELnc\",\n",
      "            \"https://github.com/WGLab/lncScore\",\n",
      "            \"https://github.com/congyingnan/TF-IDF\",\n",
      "            \"https://github.com/shaze/wcdest/\",\n",
      "            \"https://github.com/Huiyang520/DMk-BKmeans\",\n",
      "            \"https://github.com/soedinglab/kClust\",\n",
      "            \"https://github.com/cyinbox/PPI\",\n",
      "            \"https://github.com/aziele/alfpy\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13059-017-1382-0\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/theislab/Scanpy\",\n",
      "            \"https://github.com/theislab/anndata\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170505_seurat\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170503_zheng17\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170522_visualizing_one_million_cells\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170502_haghverdi16\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170430_krumsiek11\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170529_images\",\n",
      "            \"https://github.com/linnarsson-lab/loompy\",\n",
      "            \"https://github.com/linnarsson-lab/loompy\",\n",
      "            \"https://github.com/PMBio/scLVM\",\n",
      "            \"https://github.com/YosefLab/FastProject\",\n",
      "            \"https://github.com/AllonKleinLab/SPRING\",\n",
      "            \"https://github.com/dimenwarper/scimitar\",\n",
      "            \"https://github.com/jacoblevine/PhenoGraph\",\n",
      "            \"https://github.com/asncd/MIMOSCA\",\n",
      "            \"https://github.com/vals/umis\",\n",
      "            \"https://github.com/GGiecold/ECLAIR\",\n",
      "            \"https://github.com/yeolab/flotilla\",\n",
      "            \"https://github.com/theislab/scanpy\",\n",
      "            \"https://github.com/theislab/anndata\",\n",
      "            \"https://github.com/theislab/scanpy_usage\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170503_zheng17\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170505_seurat\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170522_visualizing_one_million_cells\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170502_haghverdi16\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170430_krumsiek11\",\n",
      "            \"https://github.com/theislab/scanpy_usage/tree/master/170529_images\",\n",
      "            \"https://github.com/DmitryUlyanov/Multicore-TSNE\",\n",
      "            \"https://github.com/SheffieldML/GPy\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13059-018-1387-3\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/akotlar/bystro\",\n",
      "            \"https://github.com/akotlar/bystro\",\n",
      "            \"https://github.com/akotlar/bystro/tree/master/config\",\n",
      "            \"https://github.com/akotlar/bystro\",\n",
      "            \"https://github.com/wingolab-org/GenPro/blob/master/bin/mergeSnpFiles\",\n",
      "            \"https://github.com/akotlar/bystro-paper\",\n",
      "            \"https://github.com/akotlar/bystro-paper\",\n",
      "            \"https://github.com/akotlar/bystro\",\n",
      "            \"https://github.com/akotlar/bystro/blob/master/README.md\",\n",
      "            \"https://github.com/akotlar/bystro-paper\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13104-017-2616-7\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/dohalloran/fastQ_brew\",\n",
      "            \"https://github.com/ncbi/sra-tools\",\n",
      "            \"https://github.com/lh3/seqtk\",\n",
      "            \"https://github.com/dohalloran/fastQ_brew\",\n",
      "            \"https://github.com/dohalloran/fastQ_brew\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13104-017-2788-1\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/sbmlteam/sbmlpkgspec\",\n",
      "            \"https://github.com/sbmlteam/sbmlpkgspec\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13321-017-0241-z\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/girder/girder\",\n",
      "            \"https://github.com/openlink/virtuoso-opensource\",\n",
      "            \"https://github.com/openchemistry/avogadrolibs\",\n",
      "            \"https://github.com/open-source-parsers/jsoncpp\",\n",
      "            \"https://github.com/jacobwilliams/json-fortran/wiki\",\n",
      "            \"https://github.com/wadejong/NWChem-Json\",\n",
      "            \"https://github.com/wadejong/NWChemOutputToJson\",\n",
      "            \"https://github.com/openchemistry/mongochemserver\",\n",
      "            \"https://github.com/openchemistry/mongochemclient\",\n",
      "            \"https://github.com/openchemistry/mongochemdeploy\",\n",
      "            \"https://github.com/OpenChemistry/chemicaljson/blob/master/chemicaljson.md\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13321-018-0258-y\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/mordred-descriptor\",\n",
      "            \"https://github.com/mordred-descriptor\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13326-017-0136-y\": {\n",
      "        \"github\": [\n",
      "            \"http://github.com/rdfio\",\n",
      "            \"https://github.com/semsol/arc2\",\n",
      "            \"https://github.com/semsol/arc2/wiki/SPARQL-\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s13326-017-0160-y\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/information-artifact-ontology/IAO/\",\n",
      "            \"https://github.com/oxgiraldo/SMART-Protocols\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s40064-016-2897-7\": {\n",
      "        \"github\": []\n",
      "    },\n",
      "    \"10.1186/s40168-018-0420-9\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/fangly/AmpliCopyRighter\",\n",
      "            \"https://github.com/bowmanjeffs/paprica\",\n",
      "            \"https://github.com/picrust/picrust/tree/master/tutorials/picrust/_starting/_files.zip\",\n",
      "            \"https://github.com/fangly/AmpliCopyrighter\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1186/s40169-017-0177-y\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/cbg-ethz/SCITE\",\n",
      "            \"https://github.com/PMBio/scLVM\",\n",
      "            \"https://github.com/kieranrcampbell/pseudogp\",\n",
      "            \"https://github.com/alexdobin/STAR\",\n",
      "            \"https://github.com/broadinstitute/inferCNV\",\n",
      "            \"https://github.com/virajbdeshpande/AmpliconArchitect\",\n",
      "            \"https://github.com/etal./cnvkit\",\n",
      "            \"https://github.com/ChenMengjie/SynthEx\",\n",
      "            \"https://github.com/PMBio/deepcpg\",\n",
      "            \"https://github.com/lanagarmire/SSrGE\",\n",
      "            \"https://github.com/VRam142/combinatorialHiC\",\n",
      "            \"https://github.com/ChrisMaherLab/INTEGRATE-Neo\",\n",
      "            \"https://github.com/seandavi/awesome-single-cell\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.12688/f1000research.10137.1\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/broadinstitute/wdl\",\n",
      "            \"https://github.com/ga4gh/tool-registry-schemas\",\n",
      "            \"https://github.com/ga4gh/tool-registry-schemas\",\n",
      "            \"https://github.com/common-workflow-language/cwltool\",\n",
      "            \"https://github.com/broadinstitute/cromwell\",\n",
      "            \"https://github.com/ga4gh/tool-registry-schemas\",\n",
      "            \"https://github.com/bioboxes/rfc\",\n",
      "            \"https://github.com/ga4gh/dockstore\",\n",
      "            \"https://github.com/ga4gh/dockstore-ui\",\n",
      "            \"https://github.com/broadinstitute/wdl\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.12688/f1000research.12332.1\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/TrainTheTrainer/EXCELERATE-TtT\",\n",
      "            \"https://github.com/TrainTheTrainer/EXCELERATE-TtT\",\n",
      "            \"https://github.com/ekorpela/cloud-vm-workshop/\",\n",
      "            \"https://github.com/ekorpela/cloud-vm-workshop/\",\n",
      "            \"https://github.com/carpentries/policies/blob/master/trainer-agreement.md\",\n",
      "            \"https://github.com/carpentries/policies/blob/master/trainer-agreement.md\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.12688/f1000research.12516.1\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/NCBI-Hackathons/svcompare\",\n",
      "            \"https://github.com/NCBI-Hackathons/svcompare\",\n",
      "            \"https://github.com/NCBI-Hackathons/svcompare\",\n",
      "            \"https://github.com/genome-in-a-bottle\",\n",
      "            \"https://github.com/ga4gh/benchmarking-tools/tree/master/resources/stratification-bed-files\",\n",
      "            \"https://github.com/NCBI-Hackathons/svcompare\",\n",
      "            \"https://github.com/NCBI-Hackathons/svcompare.gi\",\n",
      "            \"https://github.com/genome-in-a-bottle\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.12688/f1000research.12874.1\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/veg/hyphy.git\",\n",
      "            \"https://github.com/stamatak/standard-RAxML\",\n",
      "            \"https://github.com/clauswilke/proteinER/tree/master/src\",\n",
      "            \"https://github.com/clauswilke/proteinER/\",\n",
      "            \"https://github.com/clauswilke/proteinER/tree/master/measuring_aa_rates\",\n",
      "            \"https://github.com/clauswilke/proteinER/tree/master/measuring_dNdS\",\n",
      "            \"https://github.com/clauswilke/proteinER/tree/master/measuring_structural_features\",\n",
      "            \"https://github.com/clauswilke/proteinER/tree/master/map_structural_features\",\n",
      "            \"https://github.com/sjspielman/phyphy\",\n",
      "            \"https://github.com/clauswilke/proteinER\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.12688/f1000research.13511.2\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/BIMSBbioinfo/netSmooth\",\n",
      "            \"https://github.com/BIMSBbioinfo/netSmooth\",\n",
      "            \"https://github.com/pkathail/magic\",\n",
      "            \"https://github.com/Vivianstats/scImpute\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pcbi.1005642\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/ahgroup/DSAIDE\",\n",
      "            \"https://github.com/ahgroup/DSAIDE\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pcbi.1005740\": {\n",
      "        \"github\": []\n",
      "    },\n",
      "    \"10.1371/journal.pcbi.1005871\": {\n",
      "        \"github\": [\n",
      "            \"http://github.com/azet/community_bash_style_guide\",\n",
      "            \"http://github.com/awesome-lists/awesome-bash\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pcbi.1005968\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/reactome/graph-importer\",\n",
      "            \"https://github.com/reactome/graph-core\",\n",
      "            \"https://github.com/reactome/content-service\",\n",
      "            \"https://github.com/reactome/graph-core\",\n",
      "            \"https://github.com/reactome\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pone.0175480\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/dohalloran/phylo-node\",\n",
      "            \"https://github.com/dohalloran/phylo-node\",\n",
      "            \"https://github.com/dohalloran/phylo-node\",\n",
      "            \"https://github.com/dohalloran/phylo-node\",\n",
      "            \"https://github.com/nodejs\",\n",
      "            \"https://github.com/dohalloran/phylo-node/blob/master/README.md\",\n",
      "            \"https://github.com/ddarriba/prottest3\",\n",
      "            \"https://github.com/ddarriba/jmodeltest2\",\n",
      "            \"https://github.com/dohalloran/phylo-node\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pone.0176334\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/Bohdan-Khomtchouk/shinyheatmap\",\n",
      "            \"https://github.com/Bohdan-Khomtchouk/fastheatmap\",\n",
      "            \"https://github.com/Bohdan-Khomtchouk/shinyheatmap\",\n",
      "            \"https://github.com/Bohdan-Khomtchouk/fastheatmap\",\n",
      "            \"https://github.com/Bohdan-Khomtchouk/shinyheatmap\",\n",
      "            \"https://github.com/Bohdan-Khomtchouk/fastheatmap\",\n",
      "            \"https://github.com/Bohdan-Khomtchouk/shinyheatmap\",\n",
      "            \"https://github.com/Bohdan-Khomtchouk/fastheatmap\",\n",
      "            \"https://github.com/talgalili/heatmaply/issues/20\",\n",
      "            \"https://github.com/Bohdan-Khomtchouk/fastheatmap\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pone.0187457\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/5c077/ExpressionDB\",\n",
      "            \"https://github.com/5c077/ExpressionDB/wiki/User&#x02019;s-Guide\",\n",
      "            \"https://github.com/5c077/ExpressionDB\",\n",
      "            \"https://github.com/5c077/ExpressionDB/wiki/User&#x02019;s-Guide\",\n",
      "            \"https://github.com/5c077/ExpressionDB\",\n",
      "            \"https://github.com/5c077/ExpressionDB/wiki/User&#x02019;s-Guide\",\n",
      "            \"https://github.com/5c077/ExpressionDB/wiki/User&#x02019;s-Guide\",\n",
      "            \"https://github.com/5c077/ExpressionDB/tree/master/data\",\n",
      "            \"https://github.com/5c077/ExpressionDB/tree/master/data\",\n",
      "            \"https://github.com/5c077/ExpressionDB/tree/master/data\",\n",
      "            \"https://github.com/5c077/ExpressionDB/wiki/User&#x02019;s-Guide\",\n",
      "            \"https://github.com/5c077/ExpressionDB/issues\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pone.0188244\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/lsw5077/Pheasant-Habitat-Simulator\",\n",
      "            \"https://github.com/lsw5077/Pheasant-Habitat-Simulator\",\n",
      "            \"https://github.com/lsw5077/Pheasant-Habitat-Simulator\",\n",
      "            \"https://github.com/lsw5077/Pheasant-Habitat-Simulator\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pone.0188511\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/singularityware/singularity-python\",\n",
      "            \"https://github.com/\",\n",
      "            \"https://github.com/singularityware/singularity-python/blob/master/examples/shub/paper/compare_builds.ipynb\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pone.0189334\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer\",\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer\",\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer\",\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer#othercnvformats\",\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer#binDir\",\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer#outputFormat\",\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer/blob/master/img/rColor.txt\",\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer\",\n",
      "            \"https://github.com/FJD-CEPH/aCNViewer\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pone.0191298\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/igm-team/orion-public\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1371/journal.pone.0192520\": {\n",
      "        \"github\": []\n",
      "    },\n",
      "    \"10.1371/journal.pone.0192858\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/pmadanecki/htdp\",\n",
      "            \"https://github.com/pmadanecki/htdp\",\n",
      "            \"https://github.com/pmadanecki/htdp\",\n",
      "            \"https://github.com/pmadanecki/htdp\",\n",
      "            \"https://github.com/pmadanecki/htdp\",\n",
      "            \"https://github.com/pmadanecki/htdp\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.1523/eneuro.0233-17.2017\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com\",\n",
      "            \"https://github.com\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.2174/1874104501711010212\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/shamsaraj/ezqsar\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.3389/fninf.2017.00032\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/FNNDSC/viewerjs\",\n",
      "            \"https://github.com/FNNDSC/fmjs\",\n",
      "            \"https://github.com/FNNDSC/gcjs\",\n",
      "            \"https://github.com/FNNDSC\",\n",
      "            \"https://github.com/xtk/X\",\n",
      "            \"https://github.com/FNNDSC/AMI\",\n",
      "            \"https://github.com/FNNDSC/toolbarjs\",\n",
      "            \"https://github.com/FNNDSC/rboxjs\",\n",
      "            \"https://github.com/FNNDSC/rendererjs\",\n",
      "            \"https://github.com/FNNDSC/thbarjs\",\n",
      "            \"https://github.com/chafey/dicomParser\",\n",
      "            \"https://github.com/FNNDSC/medview\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.3389/fphar.2017.00377\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/opentox/nano-lazar-paper\",\n",
      "            \"https://github.com/opentox/lazar\",\n",
      "            \"https://github.com/enanomapper/nano-lazar\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/import.rb#L9-L122\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/model.rb#L191-L272\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/nanoparticle.rb#L22-L29\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/feature_selection.rb#L7-L34\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/similarity.rb#L22-L27\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/similarity.rb#L50-L66\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/model.rb#L234-L255\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/regression.rb#L7-L21\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/caret.rb#L8-L86\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/caret.rb#L8-L86\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/caret.rb#L59-L71\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/model.rb#L191-L272\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/caret.rb#L59-L71\",\n",
      "            \"https://github.com/opentox/lazar/blob/nano-lazar-paper.revision/lib/crossvalidation.rb#L100-L113\",\n",
      "            \"https://github.com/enanomapper/nano-lazar-paper/blob/nano-lazar-paper.revision/scripts/cv-statistics.rb\",\n",
      "            \"https://github.com/opentox/lazar\",\n",
      "            \"https://github.com/enanomapper/nano-lazar\",\n",
      "            \"https://github.com/opentox/nano-lazar-paper\",\n",
      "            \"https://github.com/enanomapper/nano-lazar-paper/tree/nano-lazar-paper.revision/figures\",\n",
      "            \"https://github.com/enanomapper/nano-lazar-paper/blob/nano-lazar-paper.revision/results/\",\n",
      "            \"https://github.com/enanomapper/nano-lazar-paper/tree/nano-lazar-paper.revision/results\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.3390/ijms18112400\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/pufengdu/UltraPse\",\n",
      "            \"https://github.com/pufengdu/UltraPse\",\n",
      "            \"https://github.com/pufengdu/UltraPse\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.3732/apps.1700051\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/joycechery/Sapindaceae\",\n",
      "            \"https://github.com/joycechery/Sapindaceae/tree/master/TranscriptomeProcessing\",\n",
      "            \"https://github.com/joycechery/Sapindaceae/blob/master/TranscriptomeProcessing/scrubReads.sh\",\n",
      "            \"https://github.com/MVZSEQ/SCPP/blob/master/2-scrubReads.pl\",\n",
      "            \"https://github.com/joycechery/Sapindaceae/blob/master/AlignReadstoCDS/AlignFix.sh\",\n",
      "            \"https://github.com/joycechery/Sapindaceae/blob/master/AlignReadstoCDS/AlignFix.pl\",\n",
      "            \"https://github.com/joycechery/Sapindaceae/blob/master/FilterSequences/BlastToRemove_Paralogs\",\n",
      "            \"https://github.com/joycechery/Sapindaceae/blob/master/FilterSequences/BlastToRemove_nonnuclear\",\n",
      "            \"https://github.com/joycechery/Sapindaceae/blob/master/FilterSequences/AverageGeneCoverage.pl\",\n",
      "            \"https://github.com/chodon/zingiberales/blob/master/2MusaCDSprocess/parse_gff_to_bed.pl\",\n",
      "            \"https://github.com/joycechery/Sapindaceae/blob/master/ASTRAL-II/ASTRAL-II-Commands\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7554/elife.32822\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/greenelab/scihub/issues/39\",\n",
      "            \"https://github.com/greenelab/library-access/issues/23\",\n",
      "            \"https://github.com/greenelab/library-access\",\n",
      "            \"https://github.com/greenelab/scihub-manuscript/issues/17\",\n",
      "            \"https://github.com/greenelab/scihub-manuscript/issues/20#issuecomment-327854780\",\n",
      "            \"https://github.com/greenelab/scihub-manuscript/issues/35#issuecomment-355171030\",\n",
      "            \"https://github.com/greenelab/scihub\",\n",
      "            \"https://github.com/greenelab/scihub-browser-data\",\n",
      "            \"https://github.com/greenelab/scihub-manuscript\",\n",
      "            \"https://github.com/greenelab/scihub/issues/9\",\n",
      "            \"https://github.com/greenelab/crossref/issues/3\",\n",
      "            \"https://github.com/greenelab/crossref\",\n",
      "            \"https://github.com/greenelab/scihub/issues/7\",\n",
      "            \"https://github.com/dhimmel/scopus\",\n",
      "            \"https://github.com/greenelab/scihub/issues/2\",\n",
      "            \"https://github.com/greenelab/scihub-manuscript/issues/36\",\n",
      "            \"https://github.com/greenelab/opencitations\",\n",
      "            \"https://github.com/greenelab/crossref\",\n",
      "            \"https://github.com/elifesciences-publications/crossref\",\n",
      "            \"https://github.com/greenelab/scihub\",\n",
      "            \"https://github.com/elifesciences-publications/scihub\",\n",
      "            \"https://github.com/greenelab/scihub-manuscript\",\n",
      "            \"https://github.com/elifesciences-publications/scihub-manuscript\",\n",
      "            \"https://github.com/greenelab/scihub-browser-data\",\n",
      "            \"https://github.com/elifesciences-publications\",\n",
      "            \"https://github.com/dhimmel/scopus\",\n",
      "            \"https://github.com/elifesciences-publications/scopus\",\n",
      "            \"https://github.com/greenelab/library-access\",\n",
      "            \"https://github.com/elifesciences-publications/library-access\",\n",
      "            \"https://github.com/greenelab/scihub-manuscript/milestone/2\",\n",
      "            \"https://github.com/greenelab/scihub-manuscript/issues/35\",\n",
      "            \"https://github.com/greenelab/scihub/blob/ca4d523e149f30be7fd3d3ae6551a26d1c625313/01.catalog-dois.ipynb\",\n",
      "            \"https://github.com/pandas-dev/pandas/blob/v0.20.1/pandas/core/algorithms.py#L399\",\n",
      "            \"https://github.com/greenelab/scihub/blob/ca4d523e149f30be7fd3d3ae6551a26d1c625313/data/doi.tsv.xz\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7717/peerj.3893\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/WGS-standards-and-analysis/datasets\",\n",
      "            \"https://github.com/WGS-standards-and-analysis/datasets\",\n",
      "            \"https://github.com/globalmicrobialidentifier-WG3/datasets\",\n",
      "            \"https://github.com/WGS-standards-and-analysis/datasets\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7717/peerj.3948\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/cgwyx/BGDMdocker.git\",\n",
      "            \"https://github.com/cgwyx/BGDMdocker/archive/master.zip\",\n",
      "            \"https://github.com/cgwyx/BGDMdocker\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7717/peerj.4013\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/pierrepo/PBxplore\",\n",
      "            \"https://github.com/pierrepo/PBxplore\",\n",
      "            \"https://github.com/pierrepo/PBxplore\",\n",
      "            \"https://github.com/pierrepo/PBxplore/tree/master/demo_paper\",\n",
      "            \"https://github.com/pierrepo/PBxplore\",\n",
      "            \"https://github.com/pierrepo/PBxplore\",\n",
      "            \"https://github.com/pierrepo/PBxplore\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7717/peerj.4088\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/danforthcenter/plantcv-v2-paper\",\n",
      "            \"https://github.com/danforthcenter/plantcv\",\n",
      "            \"https://github.com/danforthcenter/plantcv-v2-paper\",\n",
      "            \"https://github.com/demotu/BMC\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7717/peerj.4210\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/tseemann/snippy\",\n",
      "            \"https://github.com/tseemann/mlst\",\n",
      "            \"https://github.com/tseemann/abricate\",\n",
      "            \"https://github.com/sanger-pathogens/Roary\",\n",
      "            \"https://github.com/drpowell/FriPan\",\n",
      "            \"https://github.com/kwongj/roary2fripan\",\n",
      "            \"https://github.com/tseemann/snippy\",\n",
      "            \"https://github.com/kwongj/trim-aln\",\n",
      "            \"https://github.com/kwongj/cfml-maskrc\",\n",
      "            \"https://github.com/kwongj/contig-puller\",\n",
      "            \"https://github.com/tseemann/nullarbor\",\n",
      "            \"https://github.com/tseemann\",\n",
      "            \"https://github.com/kwongj\",\n",
      "            \"https://github.com/tseemann\",\n",
      "            \"https://github.com/kwongj\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7717/peerj.4262\": {\n",
      "        \"github\": [\n",
      "            \"http://github.com/jzhangc/RBiomirGS\",\n",
      "            \"http://github.com/jzhangc/RBiomirGS\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7717/peerj.4312\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/jni/skan\",\n",
      "            \"https://github.com/jni/skan\",\n",
      "            \"https://github.com/jni/skan-scripts\",\n",
      "            \"https://github.com/jni/skan\"\n",
      "        ]\n",
      "    },\n",
      "    \"10.7717/peerj.4339\": {\n",
      "        \"github\": [\n",
      "            \"https://github.com/sjspielman/leisr_validation\",\n",
      "            \"https://github.com/sjspielman/leisr_validation\",\n",
      "            \"https://github.com/sjspielman/leisr_validation\",\n",
      "            \"https://github.com/sjspielman/leisr_validation\"\n",
      "        ]\n",
      "    }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "with open('dict_of_papers.json', 'w') as outfile:  \n",
    "    json.dump(dict_of_papers, outfile)\n",
    "\n",
    "print(json.dumps(dict_of_papers, sort_keys=True, indent=4))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
